### PR TITLE
Improves download experience for export of CVS on remote R

### LINKED ISSUE #2574: Files downloaded from remote must be compressed whenever possible.
We compress files when we are sending to to RHost. We should compress when we download. Example, CVS.

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #2574

### CODE REVIEW

File: src/Host/Client/Test/RtvsPackage/BlobsTest.cs [hunk 1]
```diff
@@ -175,5 +176,84 @@ public class BlobsTest : IAsyncLifetime {
...
+                createCompressedResult.Result.Should().NotBeNull();
+
+                var blobId = ((JValue)createResult.Result).Value<ulong>();
+                var expectedData = await eval.EvaluateAsync<byte[]>($"rtvs:::get_blob({blobId})", REvaluationKind.Normal);
+
+                var blobId2 = ((JValue)createCompressedResult.Result).Value<ulong>();
+                var compressedData = await eval.EvaluateAsync<byte[]>($"rtvs:::get_blob({blobId2})", REvaluationKind.Normal);
+
+                // Decompress and validate
+                var fs = new FileSystem();
```
Comment (huguesv) [inline src/Host/Client/Test/RtvsPackage/BlobsTest.cs]: this block can be extracted and reused

File: src/Host/Client/Impl/Blobs/DataTransferSession.cs [hunk 2]
```diff
@@ -82,6 +82,27 @@ public class DataTransferSession : IDisposable {
...
+        /// <param name="filePath">Path to the file where the retrieved data will be written.</param>
+        /// <param name="doCleanUp">true to add blob upon transfer for cleanup on dispose, false to ignore it after transfer.</param>
+        public async Task FetchCompressedFileAsync(IRBlobInfo blob, string filePath, bool doCleanUp = true, IProgress<long> progress = null) {
+            string tempFilePath = _fs.GetTempFileName();
+            using (RBlobStream blobStream = await RBlobStream.OpenAsync(blob, _blobService))
+            using (Stream fileStream = _fs.CreateFile(tempFilePath)) {
+                await blobStream.CopyToAsync(fileStream, progress);
+            }
+
+            _fs.DecompressFile(tempFilePath, filePath);
```
Comment (huguesv) [inline src/Host/Client/Impl/Blobs/DataTransferSession.cs]: try-finally to ensure tempFilePath is deleted if decompress fails

File: src/Host/Client/Test/RtvsPackage/BlobsTest.cs [hunk 3]
```diff
@@ -175,5 +176,85 @@ public class BlobsTest : IAsyncLifetime {
...
+            }
+        }
+
+        private byte[] DecompressData(byte[] compressedData) {
+            byte[] decompressedData = null;
+            var fs = new FileSystem();
+            string tempFile = fs.GetTempFileName();
+            string dataFile = fs.GetTempFileName();
+            try {
+                fs.FileWriteAllBytes(tempFile, compressedData);
```
Comment (AlexanderSher) [inline src/Host/Client/Test/RtvsPackage/BlobsTest.cs]: If you want, you can implement `DeployFilesFixture` that will allow you to store files as test artifacts

Comment (karthiknadig) [inline src/Host/Client/Test/RtvsPackage/BlobsTest.cs]: I decided to remove the need for intermediate temporary files. will update the PR.

**Comment by karthiknadig** [comment]
Ping

File: src/Host/Client/Test/RtvsPackage/BlobsTest.cs [hunk 4]
```diff
@@ -175,5 +176,64 @@ public class BlobsTest : IAsyncLifetime {
...
+                var createResult = await eval.EvaluateAsync("rtvs:::create_blob(raw())", REvaluationKind.Normal);
+                createResult.Result.Should().NotBeNull();
+
+                var blobId = ((JValue)createResult.Result).Value<ulong>();
+                var actualData = await eval.EvaluateAsync<byte[]>($"rtvs:::get_blob({blobId})", REvaluationKind.Normal);
+
+                byte[] expectedData = { };
+                actualData.Should().Equal(expectedData);
+
+                await eval.ExecuteAsync($"rtvs:::destroy_blob({blobId})");
```
Comment (MikhailArkhipov) [inline src/Host/Client/Test/RtvsPackage/BlobsTest.cs]: Invariant for FxCop sake

**Comment by MikhailArkhipov** [comment]
👍

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**